<launch>
  <!-- Arguments ：机器人模型、地图文件、打开rviz、机器人运动-->
  <arg name="model" default="$(env TURTLEBOT3_MODEL)" doc="model type [burger, waffle, waffle_pi]"/>
  <arg name="map_file" default="$(find ros_tracker)/worlds/custom_map/mymap.yaml"/>
  <arg name="open_rviz" default="false"/>
  <arg name="move_forward_only" default="false"/>
  <arg name="sensor_range" default="1.0"/>
  <arg name="cmd_vel_topic" default="/cmd_vel" />
  <arg name="odom_topic" default="/odom" />
  <arg name="scan_topic" default="/scan"/>
 
  <!-- Map server ：加载地图文件 -->
  <node pkg="map_server" name="map_server" type="map_server" args="$(arg map_file)"/>
  
  <!-- AMCL ：定位 -->
  <include file="$(find turtlebot3_navigation)/launch/amcl.launch"/>
  <node name="robot_state_publisher" pkg="robot_state_publisher" type="robot_state_publisher" />
  <!-- <node pkg="tf2_ros" type="static_transform_publisher" name="static_transform_publisher" args="-0.064 0 0.122 0 0 0 base_link base_scan" /> -->
  <!-- move_base ：机器人运动-->
    <!--move_base参数配置http://wiki.ros.org/move_base -->
  <include file="$(find turtlebot3_navigation)/launch/move_base.launch"/>
  <!-- <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
    <param name="base_local_planner" value="dwa_local_planner/DWAPlannerROS" />
    <rosparam file="$(find turtlebot3_navigation)/param/costmap_common_params_$(arg model).yaml" command="load" ns="global_costmap" />
    <rosparam file="$(find turtlebot3_navigation)/param/costmap_common_params_$(arg model).yaml" command="load" ns="local_costmap" />
    <rosparam file="$(find turtlebot3_navigation)/param/local_costmap_params.yaml" command="load" />
    <rosparam file="$(find turtlebot3_navigation)/param/global_costmap_params.yaml" command="load" />
    <rosparam file="$(find turtlebot3_navigation)/param/move_base_params.yaml" command="load" />
    <rosparam file="$(find turtlebot3_navigation)/param/dwa_local_planner_params_$(arg model).yaml" command="load" />
    <remap from="cmd_vel" to="$(arg cmd_vel_topic)"/>
    <remap from="odom" to="$(arg odom_topic)"/>
    <remap from="scan" to="$(arg scan_topic)"/>

  </node> -->
  <!-- rviz ：打开riviz-->
  <group if="$(arg open_rviz)"> 
    <node pkg="rviz" type="rviz" name="rviz" required="true"
          args="-d $(find turtlebot3_navigation)/rviz/turtlebot3_navigation.rviz"/>
  </group>
</launch>